src/fhirtypes/common.ts

```diff
@@ -713,3 +713,26 @@ export class HasId {
     }
   }
 }
+
+/**
+ * Checks if a provided type can be treated as a Reference
+ * @param type - The type being checked
+ * @returns - True if the type can be treated as a reference, false otherwise
+ */
+export function isReferenceType(type: string): boolean {
+  return ['Reference', 'CodeableReference'].includes(type);
+}
+
+/**
+ * Gets the ElementDefinition that is providing reference constraints on a given Reference type element.
+ * The current application of this is to get a CodeableReference element that may be constraining its CodeableReference.reference
+ * element
+ * @param ed - The Reference type ElementDefinition.
+ * @returns - The element which contains the type constraints that apply to the Reference type element. If
+ * the element is not of Reference type, it is just returned.
+ */
+export function getReferenceConstrainingElement(ed: ElementDefinition) {
+  return ed.type[0].code === 'Reference' && ed.parent()?.type?.[0]?.code === 'CodeableReference'
+    ? ed.parent()
+    : ed;
```

Comment:
The function is actually intended to _not_ assume that the passed in element is a reference. At the point where this function is used, the element has an unknown type. The type is checked in `assignFHIRValue`. I think putting it in a function was just confusing. I will de-function it.

Reply:
OK, I think the name "getReferenceConstrainingElement" confused me since it can return an element that doesn't constrain a reference.

Good riddance. ;-)